tests/Struct/FindologicProductTest.php

```diff
@@ -186,6 +187,95 @@ public function testNoProductCategories(): void
         );
     }
 
+    /**
+     * @dataProvider categoryAndCatUrlWithIntegrationTypeProvider
+     */
+    public function testOnlyUniqueCategoriesExported(
+        ?string $integrationType,
+        array $categories,
+        array $expectedCatUrls
+    ): void {
+        foreach ($categories as $key => $category) {
+            $navigationCategoryId = $this->salesChannelContext->getSalesChannel()->getNavigationCategoryId();
+            $categories[$key]['parentId'] = $navigationCategoryId;
+        }
+
+        $productEntity = $this->createTestProduct(['categories' => $categories]);
+        $config = $this->getMockedConfig($integrationType);
+        $findologicProductFactory = new FindologicProductFactory();
+        $findologicProduct = $findologicProductFactory->buildInstance(
+            $productEntity,
+            $this->router,
+            $this->getContainer(),
+            $this->shopkey,
+            [],
+            new XMLItem('123'),
+            $config
+        );
+        $categoriesToBeChecked = [];
+        $isUnique = true;
+        $this->assertTrue($findologicProduct->hasAttributes());
+        $attributes = $findologicProduct->getAttributes();
+        if (count($expectedCatUrls) > 0) {
+            $catValues = $attributes[1]->getValues();
+        } else {
+            $catValues = $attributes[0]->getValues();
+        }
+        foreach ($catValues as $category) {
+            if (!empty(array_search($category, $categoriesToBeChecked))) {
+                $isUnique = false;
+                break;
+            }
+            $categoriesArray[] = $category;
+        }
```

Comment:
@D3xter7 This does not guarantee that the values you are checking are of the categories. This can be any other attribute as well. Please do some valid assertions here too.